src/Filter/NestedFilter.php

```diff
@@ -0,0 +1,159 @@
+
+    public const PATH_SEPARATOR_EXPECTED = '.';
+    public const PATH_SEPARATOR = '_';
+
+    /** @var FilterInterface */
+    private $wrappedFilter;
+
+    public static function new(string $propertyName, string $label = null): self
+    {
+        throw new \RuntimeException('Instead of this method, use the "wrap()" method.');
+    }
+
+    public static function wrap(FilterInterface $filter): FilterInterface
+    {
+        $filterDto = $filter->getAsDto();
+        $property = $filterDto->getProperty();
+
+        if (false === strpos($property, self::PATH_SEPARATOR_EXPECTED)) {
+            return $filter;
+        }
+
+        return (new self())
+            ->setFilterFqcn(__CLASS__)
+            ->setProperty($property)
+            ->setFormType($filterDto->getFormType())
+            ->setFormTypeOptions($filterDto->getFormTypeOptions())
+            ->setWrappedFilter($filter)
+        ;
+    }
+
+    public function apply(QueryBuilder $queryBuilder, FilterDataDto $filterDataDto, ?FieldDto $fieldDto, EntityDto $entityDto): void
+    {
+        $propertyPath = $filterDataDto->getProperty();
+
+        [$targetClassMetadata, $targetProperty] = self::extractTargets(
+            $queryBuilder->getEntityManager(),
+            $entityDto->getFqcn(),
+            $propertyPath
+        );
+
+        $wrappedEntityDto = new EntityDto($targetClassMetadata->getName(), $targetClassMetadata);
+        $wrappedFilter = $this->getWrappedFilter();
+        $wrappedFilterDto = $wrappedFilter->getAsDto();
+        $wrappedFilterDto->setProperty($targetProperty);
+
+        // Apply required left joins and get the alias we have to work with
+        $alias = $this->applyLeftJoins($queryBuilder, $filterDataDto->getEntityAlias(), $propertyPath);
+
+        // Recreate FilterDataDto adapted for the wrapped filter
+        $wrappedFilterDataDto = FilterDataDto::new($filterDataDto->getIndex(), $wrappedFilterDto, $alias, [
+            'value' => $filterDataDto->getValue(),
+            'value2' => $filterDataDto->getValue2(),
+            'comparison' => $filterDataDto->getComparison(),
+        ]);
+
+        $wrappedFilterDto->apply($queryBuilder, $wrappedFilterDataDto, null, $wrappedEntityDto);
+    }
+
+    public static function extractTargets(ObjectManager $objectManager, string $class, string $propertyPath): array
+    {
+        $segments = explode(self::PATH_SEPARATOR, $propertyPath);
+        $metadata = $objectManager->getClassMetadata($class);
+        $lastIndex = \count($segments) - 1;
+        $property = null;
+
+        foreach ($segments as $i => $prop) {
+            if (!$metadata->hasField($prop) && !$metadata->hasAssociation($prop)) {
+                self::throwInvalidPropertyPathException($propertyPath, $class);
+            }
+
+            // The target property must be at the end of path
+            if ($i === $lastIndex) {
+                $property = $prop;
+                break;
+            }
```

Comment:
This adds support for embeddables.

```suggestion
        $prefix = '';

        foreach ($segments as $i => $prop) {
            $prop = $prefix.$prop;

            if (!$metadata->hasField($prop) && !$metadata->hasAssociation($prop)) {
                self::throwInvalidPropertyPathException($propertyPath, $class);
            }

            // The target property must be at the end of path
            if ($i === $lastIndex) {
                $property = $prop;
                break;
            }

            if ($metadata->hasField($prop)) {
                $prefix .= $prop.'.';
                continue;
            }
```